components/cam/src/physics/cam/debug_info.F90

```diff
@@ -0,0 +1,81 @@
+module debug_info
+
+  !-------------------------------------------------------------
+  ! This is a light weight module to report parameters which
+  ! might be of interest in case of a simulation crash.
+  !
+  ! Author: Balwinder Singh
+  ! (Following suggestions from Peter Caldwell and Aaron Donahue)
+  !-------------------------------------------------------------
+
+  implicit none
+  private
+
+  !Variables updated during the simulation
+  integer :: macmiciter, chunk
+  !$OMP THREADPRIVATE(macmiciter,chunk )
+
+  public:: &
+       get_debug_chunk, &       !update chunk info from the simulation
+       get_debug_macmiciter, &  !update macro/microphysics iteration count
+       report_error_info        !report info on the current lat/lon and other simulation parameters
+
+contains
+
+  subroutine get_debug_chunk(chunk_in)
+
+    !Obtain chunk # from tphycbc
+
+    implicit none
+
+    integer :: chunk_in
+
+    chunk = chunk_in
+
+  end subroutine get_debug_chunk
+
+
+  subroutine get_debug_macmiciter(iter)
+
+    !Obtain iteration # of macro/microphysics sub-stepping iteration
+
+    implicit none
+
+    integer, intent(in) :: iter
+
+    macmiciter = iter
+
+  end subroutine get_debug_macmiciter
+
+
+  subroutine report_error_info(cause, calling_func, icol, klev)
+
+    !Reports simulation parameters
+    use cam_logfile,    only: iulog
+    use phys_grid,      only: get_rlat_p, get_rlon_p
+    use time_manager,   only: get_nstep
+    use physics_utils,  only: rtype
+    implicit none
+
+    integer, intent(in) :: icol !column(lat/lon) index
+    integer, optional, intent(in) :: klev !vertical index
+    character(len=*), intent(in) :: cause, calling_func
+
+    !local variables
+    real(rtype), parameter::  radian2deg = 180.0_rtype/(4.0_rtype*atan(1.0_rtype)) !180/pi = 57.296
+
+    write(iulog,'(a)') '*****************************************************************'
+    write(iulog,'(a)')'Fail due to '//trim(adjustl(cause))//' in subroutine/function:'//trim(adjustl(calling_func))
+    write(iulog,'(a,f12.8,a,f12.8,a)')'Column location in degrees:(lat,lon) is:(',get_rlat_p(chunk,icol)*radian2deg,',', &
+         get_rlon_p(chunk,icol)*radian2deg,')'
+    write(iulog,'(a,i5)')'macmic_it is:', macmiciter
```

Comment:
the `macmiciter` information is really only useful in cases where macro or micro physics is the cause of the error.  In keeping with general purpose use of this debug info do we want to have macmiciter be printed as an optional argument?  Something like `logical, intent(in), optional :: print_macmic` and then have `if (present(print_macmic))` here?

Again, really only a question.  When it comes to code failures we may not be bothered by extra info that may not be relevant if the code errors out during deep convection say, or radiation.

Reply:
For the HR compset there really isn't much other than mac and mic. This is a good suggestion, but since we're trying to create a quick stopgap utility until we abandon F90, I don't think the extra generality is warranted.

Reply:
Peter, Fair point.  That was my thinking too, but just in case we were planning on keeping this functionality for longer-term I thought I would bring it up.  Another counter-point to my original point is that if the only people who will use this function are fellow SCREAMER's than it won't confuse anyone since we are all pretty aware of what's going on.

Reply:
This is also a good suggestion and fairly easy to add.